src/components/Touchable.tsx

```diff
@@ -1,27 +1,41 @@
 import * as React from 'react'
-import { TouchableWithoutFeedbackProps, View } from 'react-native'
+import { TouchableWithoutFeedbackProps, View, ViewStyle } from 'react-native'
 import PlatformTouchable from 'react-native-platform-touchable'
 
 export interface Props extends TouchableWithoutFeedbackProps {
   borderless?: boolean
   children: React.ReactNode // must only have one direct child. see https://github.com/react-native-community/react-native-platform-touchable#touchable
   borderRadius?: number
+  touchableStyle?: ViewStyle
+  useForeground?: boolean
 }
 
 /**
  * @param borderless - If true, the touchable has a borderless ripple effect.
  * @param borderRadius - should be added if the touchable component has rounded corners, to prevent the ripple effect from going outside the component on tap (Android)
+ * @param useForeground - If present, will set foreground parameter instead of background. Helps with z-index issues within absolutely-positioned parents. See https://github.com/facebook/react-native/issues/17200
+ * @param touchableStyle - If present, will set the style on the Touchable wrapper. Used to override borderRadius in case of atypical borders.
  * @returns A touchable component
  */
-export default function Touchable({ borderless, borderRadius = 0, ...passThroughProps }: Props) {
+export default function Touchable({
+  touchableStyle,
+  useForeground,
+  borderless,
+  borderRadius = 0,
+  ...passThroughProps
+}: Props) {
   const background = borderless
     ? PlatformTouchable.SelectableBackgroundBorderless()
     : PlatformTouchable.SelectableBackground()
   return borderRadius === 0 ? (
     <PlatformTouchable {...passThroughProps} background={background} />
   ) : (
-    <View style={[{ borderRadius }, !borderless && { overflow: 'hidden' }]}>
-      <PlatformTouchable {...passThroughProps} background={background} />
+    <View style={[{ borderRadius }, touchableStyle, !borderless && { overflow: 'hidden' }]}>
```

Comment:
i have feelings about this because it feels like `touchableStyle` removes the need for `borderRadius`, but at the same time gives too much flexibility to this component (it's definitely harder to screw up the styles by passing in a number for borderRadius as opposed to any stylesheet).

it seems like you're using `touchableStyle` to undo some effects of `borderRadius`. i wonder instead if the borderRadius be a number _or_ an array (like how in css you can apply `padding` in an array)?

Reply:
Yeah, I wasn't too pleased with this either. You can't specify border radii using an array/list of values, so I've opted for a similar approach using a style object restricted to certain keys.